exodus_lambda/functions/origin_request.py

```diff
@@ -172,23 +173,41 @@ def resolve_aliases(self, uri):
     def handle_cookie_request(self, event):
         now = datetime.utcnow()
         ttl = timedelta(minutes=self.conf.get("cookie_ttl", 720))
-        domain = event["Records"][0]["cf"]["config"]["distributionDomainName"]
-        uri = event["Records"][0]["cf"]["request"]["uri"]
+        record = event["Records"][0]["cf"]
+        domain = record["config"]["distributionDomainName"]
+        uri = record["request"]["uri"]
+        querystring_dict = parse_qs(record["request"]["querystring"])
+        policy = querystring_dict.get("Policy", [None])[0]
+        signature = querystring_dict.get("Signature", [None])[0]
 
         self.logger.info("Handling cookie request: %s", uri)
 
-        signer = Signer(self.cookie_key, self.conf.get("key_id"))
+        # Only load the cookie key if we need to generate a signature.
+        cookie_key = None
+        if not all([policy, signature]):
```

Comment:
I don't think this should be conditional, as one of the major benefits of this change should be that we can completely drop the related secret (and code) from exodus-lambda, but we can't do that as long as we'd still need it in some cases.

If you had in mind that this would be used temporarily, I'd rather we just get the exodus-gw side completed first.  So, let's say exodus-gw does:

- uses a valid signed URL for `/_/cookie` (as it does today)
- but also includes extra policies for /content/ and /origin/ within that request, under a new query string parameter

That should work fine against exodus-cdn today, and the extra policies are just ignored.  Then exodus-cdn can have a single update which makes it start using the extra policies instead of generating them by itself.  We can rip out the signing/secret code in exodus-cdn immediately as part of that change, there doesn't seem to be any need to keep it around as a conditional if we have already updated exodus-gw everywhere.